lib/core/restServer.hpp

```diff
@@ -231,6 +231,9 @@ class restServer {
      */
     void remove_all_aliases();
 
+    /// The port to use
```

Comment:
Rather than expose this publicly, you might consider keeping it private and binding a public const reference to it during construction. Then it's effectively readonly publicly but can be modified internally during `start`.